### ISSUE TITLE
Implementations in nested types are not registered

### ISSUE THREAD

**Issue by seanterry** [Dreamescaper/ServiceScan.SourceGenerator#16]
I love the idea of this project. However, in trying it out I ran into a barrier for entry with my existing code: Nested types do not seem to be discovered by the scanner.

For example, consider the following validator:

```csharp
public sealed record SubmitArgs( string Value )
{
    public sealed class Validator : AbstractValidator<SubmitArgs> { /* ... */ }
}
```

Using the sample registation:
```csharp
[GenerateServiceRegistrations(AssignableTo = typeof(IValidator<>), Lifetime = ServiceLifetime.Singleton)]
public static partial IServiceCollection AddValidators(this IServiceCollection services);
```

This will result in the validator not being registered, and a DI0005 warning.

If made a non-nested type, the registration works as expected. Since our convention is to nest validators, handlers, and entity type configurations within the type they handle, recognizing nested types would make adoption significantly easier.

**Comment by Dreamescaper** [comment]
Nice catch!

Fixed in version 1.3.6 :
https://www.nuget.org/packages/ServiceScan.SourceGenerator/1.3.6

**Comment by seanterry** [comment]
Awesome turnaround time! I'll play with it a bit more.

🇺🇦
